src/unix/fs.c

```diff
@@ -217,6 +217,21 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static ssize_t uv__fs_mkstemp(uv_fs_t* req) {
+  static int no_cloexec_support;
+
+#ifdef O_CLOEXEC
+  no_cloexec_support = 1;
+#endif
+
+  if (no_cloexec_support == 1) {
```

Comment:
This detection code is not going to fly, I'm afraid. glibc added mkostemp in version 2.7, but we support 2.5 (can you configm @bnoordhuis?).

I checked a bit and turns out that some BSDs got mkostemp recently (FreeBSd 10, OpenBSD 5.7) so it's probably a safer bet to use `mkstemp` and then `fcntl` to set `O_CLOEXEC` on the fd.